src/dmd/dsymbolsem.d

```diff
@@ -3458,19 +3458,30 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
                 if (s)
                 {
-                    auto fd = s.isFuncDeclaration();
-                    assert(fd);
-
                     HdrGenState hgs;
-                    OutBuffer buf1, buf2;
+                    OutBuffer buf;
 
-                    functionToBufferFull(cast(TypeFunction)(fd.type), &buf1,
-                        new Identifier(fd.toPrettyChars()), &hgs, null);
-                    functionToBufferFull(cast(TypeFunction)(funcdecl.type), &buf2,
+                    auto fd = s.isFuncDeclaration();
+                    functionToBufferFull(cast(TypeFunction)(funcdecl.type), &buf,
                         new Identifier(funcdecl.toPrettyChars()), &hgs, null);
+                    char* funcdeclToString = buf.extractString();
```

Comment:
extractString() causes a memory leak. Use peekString(). Also declare funcdeclToString as const, and rename it to funcdeclToChars because it is a char*, not a D string.

Reply:
extractString was used before .... Chesterton's Fence.

>and rename it to funcdeclToChars

Indeed but I wanted to be consistent with `extractString`/`peekString`

Reply:
Yes, but those are named wrongly. The convention should be `String` for char[], and `Chars` for zero terminated char*.